calliope/test/test_backend_helper_functions.py

```diff
@@ -76,11 +78,61 @@ def test_squeeze_primary_carriers_not_in_model(dummy_model_data):
         squeeze_carriers(dummy_model_data.all_true_carriers, "foo")
 
 
-def test_get_connected_link(dummy_model_data):
-    get_connected_link = helper_functions.get_connected_link(dummy_model_data)
-    connected_link = get_connected_link(dummy_model_data.with_inf)
+@pytest.mark.parametrize(
+    ["lookup", "expected"],
+    [
+        (
+            {"techs": "lookup_techs"},
+            [[1.0, np.nan, np.nan, np.nan], [np.inf, np.nan, 2.0, np.nan]],
+        ),
+        (
+            {"nodes": "link_remote_nodes", "techs": "link_remote_techs"},
+            [[np.inf, np.nan, 2, np.nan], [3, np.nan, np.nan, np.nan]],
+        ),
+    ],
+)
+def test_select_from_lookup_table(dummy_model_data, lookup, expected):
```

Comment:
Yes, sorry, a docstring will go in. It implements [vectorised indexing](https://docs.xarray.dev/en/stable/user-guide/indexing.html#vectorized-indexing) but in a way that updates the values in the array without reorganising the index, e.g.:

original array:

| idx | val |
|-----|-----|
| A   | 1    |
| B   | 2    |
| C   | 3    |

Lookup array:

| idx | val |
|-----|-----|
| A   | B    |
| B   | A    |
| C   | C    |


test_select_from_lookup_table(orig, lookup, "val")

| idx | val |
|-----|-----|
| A   | 2    |
| B   | 1    |
| C   | 3    |

Is that enough of an explainer to understand for the docstring?

Reply:
Thanks for the explanation! Understood. I assume that the index of the lookup array can be anything, but the values have to exist in the index of orig?
So this lookup array would work:

| idx | val |
|-----|-----|
| A   | B    |
| B   | A    |
| C   | C    |
| D   | A    |

Reply:
No, that array would not since the two arrays are ultimately linked to the same dataset, so the index "idx" would need to be identical between the two for them to exist concurrently.